gh-103000: Optimise dataclasses asdict/astuple for common types (#103005)
python/cpython: Misc/NEWS.d/next/Library/2023-03-24-20-49-48.gh-issue-103000.6eVNZI.rst

> Performance improvement for :func:`dataclasses.astuple` and :func:`dataclasses.asdict` in cases where the contents are common python types.

Reviewer: Nit: should wrap documentation (including news entries) to <=80 chars where possible (also a few small wording tweaks):

```suggestion
Improve performance of :func:`dataclasses.astuple` and
:func:`dataclasses.asdict` in cases where the contents are common Python types.
```
